src/main/java/core/basesyntax/service/ShopService.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service;
+
+import core.basesyntax.service.filehandler.ReadFromFile;
+import core.basesyntax.service.strategy.ShopActivityStrategy;
+import java.util.Map;
+
+public class ShopService {
+    private final ReadFromFile fileInfo = new ReadFromFile();
+    private final String filePath = "src/main/resources/StorageInfo.csv";
+    private final String info = fileInfo.readFromFile(filePath);
+    private final String[] fileInfoArr = info.split("[\\r\\n\\s]+");
+
+    public void doStrategy(Map<String, ShopActivityStrategy> strategyMap) {
+        for (int i = 0; i < fileInfoArr.length; i++) {
+            String[] activityTypeArray = fileInfoArr[i].split(",");
+            String activityType = activityTypeArray[0];
+            String fruit = activityTypeArray[1];
+            int quantity = Integer.parseInt(activityTypeArray[2]);
+            ShopActivityStrategy strategy = strategyMap.get(activityType);
+            strategy.workWithActivities(fruit, quantity);
+        }
```

Comment:
I think , It will be better to transform this method and class to method and class  DataParser for example. It will take list of strings from readFromFile metod and return List of String with informationabout operations for stratage.

Reply:
I think why if I create a Parser and take String from readFromFile and return list of String with all the necessary information

Reply:
Actually Anna provided good Idea, it will be better if FileReader will return List of strings and FruitParser will work with list not with single string, it will reduce amount of code both in Reader and Parses classes